slicedimage/backends/_base.py

```diff
@@ -16,5 +19,37 @@ def write_file_from_handle(self, name, source_handle, block_size=(128 * 1024)):
             dest_handle.write(data)
 
 
-class FileNotFoundError(Exception):
-    pass
+class ChecksummingContextManager(object):
```

Comment:
A short docstring that says what this thing does would be helpful. I think I summarized it in a comment below (if not, please correct me!)

Reply:
Done-ish.  You may wish to check the language.